lsi/rom/tools/cp16mic.py

```diff
@@ -0,0 +1,981 @@
+                    continue
+                if s == ';':
+                    if pos != start:
+                        yield string[start:pos]
+                    start = -1
+                    break
+                if s == "'":
+                    quota = 1
+                elif s == '"':
+                    quota = 2
+                continue
+            if slash:
+                slash = 0
+                continue
+            if s == '\\':
+                slash = 1
+                continue
+            if quota == 1:
+                if s == "'":
+                    quota = 0
+            else:
+                if s == '"':
+                    quota = 0
+        if start >= 0:
+            yield string[start:]  
+    #
+    # Parse the optional parameters after opcode/directive
+    #
+    def parse_args(self, opcode, line):
+        for field in self.parse_gen(line):
+            field = field.strip(' \t')
+            if field and field[0] not in '\'\"':
+                field = field.upper()
+            opcode.append(field)
+        return opcode
+
+    def insert_data(self, data):
+        if not MIN_LC <= self.lc < MAX_LC:
+            raise SyntaxError("Location is out of range '0x%04X'" % self.lc)
+        if self.data[self.lc] >=0 and self.data[self.lc] != data:
+            raise SyntaxError("Location multiple commit '0x%04X'" % self.lc)
+        self.data[self.lc] = data
+        return
+
+    def do_opcode(self, opcode):
+        oc = opcode[0]
+        obj = wd16.dir_ops.get(oc.lower())
+        if obj is None:
+            raise SyntaxError("invalid command or directive '%s'" % oc[0])
+        narg = len(opcode) - 1
+        if narg < obj[1]:
+            raise SyntaxError("not enough parameters for '%s'" % oc[0])
+        if obj[0] < 0:
+            if narg > obj[1]:
+                raise SyntaxError("too many directive parameters '%s'" % oc[0])
+        elif narg > (obj[1] + 2):
+            raise SyntaxError("too many parameters for '%s'" % oc[0])
+        narg -= obj[1]
+        word = obj[0]
+        if narg >= 1:
+            #
+            # Handle the LRR, RSVC and TTL bit field
+            #
+            oc = opcode[obj[1]+1]
+            if oc and self.ps == 2:
+                ext = self.eval_exp(oc)
+                if ext & ~0x3F:
+                    raise SyntaxError("extension is out of range 0x%X" % ext)
+                word |= ext << 16
+        if narg >= 2:
+            #
+            # Handle the translation code field
+            #
+            oc = opcode[obj[1]+2]
+            if oc and self.ps == 2:
+                ext = self.symb.get(oc)
+                if ext is None or ext & ~0x7F:
+                    raise SyntaxError("invalid translation code '%s'" % oc)
+                word |= ext << 24
+        try:
+            word = obj[2](self, opcode, word)
+        except SyntaxError as err:
+            if obj[0] < 0:
+                raise SyntaxError(err)
```

Comment:
In this case you will have 2 `SyntaxError` objects, where there embedded one contains "message string" as an argument, and outer one contains `err` as `SyntaxError` instance.
It is enough just to do  `raise err` here

Reply:
nice to know, thanks